stopes/ui/seamlisten/backend/app/fileviewer.py

```diff
@@ -105,52 +103,56 @@ def open_zip_file(file: str, start_idx: int, end_idx: int) -> tp.List[LineResult
 
 
 @router.post("/annotations/")
-def get_annotations(
-    query: AnnotationQuery,
-) -> tp.List[LineResult]:
+def get_annotations(query: AnnotationQuery) -> tp.List[LineResult]:
     path = query.gz_path.strip()
-    print(path, query.start_idx, query.end_idx)
-    if path.endswith(".tsv.gz") or path.endswith(".tsv"):
+    resolved_path = Path(path).expanduser().resolve()
+
+    if resolved_path.suffixes in ([".gz"], [".tsv"]):
         try:
-            return open_segment_tsv(path, query.start_idx, query.end_idx)
+            return open_segment_tsv(str(resolved_path), query.start_idx, query.end_idx)
         except FileNotFoundError:
             raise HTTPException(
                 status_code=404,
-                detail="""
-                File not found
-            """,
+                detail="File not found",
             )
-    elif path.endswith(".zip"):
-        if not Path(path).exists():
+    elif resolved_path.suffixes == [".zip"]:
```

Comment:
you could have a `myfile.tsv.zip` in which case this would not pass but should. This would be safer:
```suggestion
    elif resolved_path.suffixes[-1] == ".zip":
```